front/components/agent_builder/capabilities/mcp/MCPServerViewsDialog.tsx

```diff
@@ -0,0 +1,694 @@
+} from "@app/lib/actions/constants";
+import { getAvatarFromIcon } from "@app/lib/actions/mcp_icons";
+import { getMCPServerRequirements } from "@app/lib/actions/mcp_internal_actions/input_configuration";
+import type { MCPServerViewType } from "@app/lib/api/mcp";
+import { useModels } from "@app/lib/swr/models";
+import { O4_MINI_MODEL_ID } from "@app/types";
+
+export type SelectedTool =
+  | {
+      type: "MCP";
+      view: MCPServerViewType;
+      configuredAction?: AgentBuilderAction;
+    }
+  | { type: "DATA_VISUALIZATION" };
+
+const DEFAULT_REASONING_MODEL_ID = O4_MINI_MODEL_ID;
+
+export type DialogMode =
+  | { type: "add" }
+  | { type: "edit"; action: AgentBuilderAction; index: number }
+  | { type: "info"; action: AgentBuilderAction };
+
+type MCPActionWithConfiguration = AgentBuilderAction & {
+  type: "MCP";
+  configuration: MCPServerConfigurationType;
+};
+
+function isMCPActionWithConfiguration(
+  action: AgentBuilderAction
+): action is MCPActionWithConfiguration {
+  return (
+    action.type === "MCP" &&
+    action.configuration !== null &&
+    action.configuration !== undefined &&
+    typeof action.configuration === "object" &&
+    "mcpServerViewId" in action.configuration
+  );
+}
+
+interface MCPServerViewsDialogProps {
+  addTools: UseFieldArrayAppend<AgentBuilderFormData, "actions">;
+  dataVisualization: ActionSpecification | null;
+  mode: DialogMode | null;
+  onModeChange: (mode: DialogMode | null) => void;
+  onActionUpdate?: (action: AgentBuilderAction, index: number) => void;
+}
+
+export function MCPServerViewsDialog({
+  addTools,
+  dataVisualization,
+  mode,
+  onModeChange,
+  onActionUpdate,
+}: MCPServerViewsDialogProps) {
+  const { owner } = useAgentBuilderContext();
+  const sendNotification = useSendNotification();
+  const { reasoningModels } = useModels({ owner });
+  const {
+    mcpServerViews: allMcpServerViews,
+    defaultMCPServerViews,
+    nonDefaultMCPServerViews,
+    isMCPServerViewsLoading,
+  } = useMCPServerViewsContext();
+
+  const isEditMode = !!mode && mode.type === "edit";
+  const isInfoMode = !!mode && mode.type === "info";
+  const isAddMode = !mode || mode.type === "add";
+
+  const [selectedToolsInDialog, setSelectedToolsInDialog] = useState<
+    SelectedTool[]
+  >([]);
+
+  const [isOpen, setIsOpen] = useState(!!mode);
+  const [currentPageId, setCurrentPageId] = useState<ConfigurationPagePageId>(
+    isEditMode
+      ? CONFIGURATION_DIALOG_PAGE_IDS.CONFIGURATION
+      : isInfoMode
+        ? CONFIGURATION_DIALOG_PAGE_IDS.INFO
+        : CONFIGURATION_DIALOG_PAGE_IDS.TOOL_SELECTION
+  );
+  const [configurationTool, setConfigurationTool] =
+    useState<AgentBuilderAction | null>(isEditMode ? mode.action : null);
+
+  const [configurationMCPServerView, setConfigurationMCPServerView] =
+    useState<MCPServerViewType | null>(null);
+  const [infoMCPServerView, setInfoMCPServerView] =
+    useState<MCPServerViewType | null>(null);
+
+  React.useEffect(() => {
```

Comment:
nit: trying to get the same way of using as `useState` and other react hooks
```suggestion
  useEffect(() => {
```